### PR TITLE
Fix Provider Issues

### PR DESCRIPTION
**Pull request by thebigsmileXD**
- [x] This PR actually submits something - don't use this system as a messaging service!
- [x] This PR isn't duplicated - you can check if it is by scanning the small list - link is on the navigation bar for this repository.
- [x] This PR includes appropriate markdown for sections - e.g. code blocks for suggested code.
- [x] This PR description and comments in code is understandable - feel free to use your native language to write if you are not comfortable with English.
- [ ] This PR has a single branch for itself in your fork - don't just commit to the master branch of your repository!
- [x] This PR has comments written in clear English - please don't write unreadable comments just for your eyes.

## Pull Request description
This fixes issues with the MySQL/MySQLi Provider not working

## Optional information
{Type something here}

### CODE REVIEW

**Comment by thebigsmileXD** [comment]
Well, this works too ;)

File: src/MyPlot/provider/MySQLiProvider.php [hunk 1]
```diff
@@ -77,20 +90,23 @@ public function deletePlot(Plot $plot): bool{
 	}
 
 	public function getPlot(string $levelName, int $X, int $Z): Plot{
-		if ($plot = $this->getPlotFromCache($levelName, $X, $Z)) {
+		if (($plot = $this->getPlotFromCache($levelName, $X, $Z)) instanceof Plot) {
```
Comment (thebigsmileXD) [inline src/MyPlot/provider/MySQLiProvider.php]: I saw several of these weird checks. As i know, if() only checks for true/false, not for null/not null

File: src/MyPlot/provider/MySQLiProvider.php [hunk 2]
```diff
@@ -77,20 +90,23 @@ public function deletePlot(Plot $plot): bool{
...
+		$stmt = $this->db->prepare("SELECT id, name, owner, helpers, denied, biome FROM plots WHERE level = ? AND X = ? AND Z = ?;");
+		$stmt->bind_param('sii', $levelName, $X, $Z);
+		$result = $stmt->execute();
+		if ($result === false) {
+			$this->plugin->getLogger()->error($stmt->error);
+		}
+		$result = $stmt->get_result();
 		if ($val = $result->fetch_array(MYSQLI_ASSOC)) {
-			if ($val["helpers"] === null or $val["helpers"] === "") {
+			if ($val["helpers"] === null or empty($val["helpers"] === "")) {
```
Comment (jasonwynn10) [inline src/MyPlot/provider/MySQLiProvider.php]: you put an empty check around a bool here

File: src/MyPlot/provider/MySQLiProvider.php [hunk 3]
```diff
@@ -77,20 +90,23 @@ public function deletePlot(Plot $plot): bool{
...
 				$helpers = [];
 			} else {
 				$helpers = explode(",", (string)$val["helpers"]);
 			}
-			if ($val["denied"] === null or $val["denied"] === "") {
+			if ($val["denied"] === null or empty($val["denied"] === "")) {
```
Comment (jasonwynn10) [inline src/MyPlot/provider/MySQLiProvider.php]: you put an empty check around a bool here also

File: .poggit.yml [hunk 4]
```diff
@@ -2,7 +2,4 @@
 projects:
   MyPlot:
     path: ""
-    libs:
-      - src: Falkirks/spoondetector/spoondetector
```
Comment (jasonwynn10) [inline .poggit.yml]: This is not happening

File: src/MyPlot/provider/MySQLiProvider.php (hunk 2 above)
Comment (thebigsmileXD) [inline src/MyPlot/provider/MySQLiProvider.php]: Ahhhrrggg 🤦‍♂️

File: .poggit.yml (hunk 4 above)
Comment (thebigsmileXD) [inline .poggit.yml]: I forgot to create a branch, i'll create a new PR anyways.

**Pull request closed without merging**